Flakey Test: Postman sync bundle api still returns unexpected RECEIVED_BUNDLE status

This relates to issue #28130 We were calling the asynchronous endpoint and as such can be expected to see a RECEIVED_BUNDLE status as we do not wait for success.

After this fix was merged and we are now using the synchronous endpoints we are still getting some test failures due to an unexpected RECEIVED_BUNDLE status.  This is due to code paths in the syncrhonous code still returning a RECIEVED_BUNDLE status occasionally rather than an expected SUCCESS.   

It has to be analyzed to see if this status should be expected in the synchronous api call and what the expected behavior is from the client in response to this, if this is expected the Postman test code uploading the bundles needs to be able to handle this response correctly rather than throwing an error.  Otherwise we need to fix the underlying code so it does not return this unexpected status if it is only expected for the async operation. 

The source of the RECEIVED_BUNDLE message is most likely to be here

(https://github.com/dotCMS/core/blob/da5e14dd638a3bc5056cc1fe72dc80f75936b7c6/dotCMS/src/main/java/com/dotcms/rest/BundleResource.java#L896)


Seen on nightly run here https://github.com/dotCMS/core/actions/runs/8856980429/job/24324051077#step:10:41537


[14_Nightly Test _ Run Postman Tests - graphql.txt](https://github.com/dotCMS/core/files/15153741/14_Nightly.Test._.Run.Postman.Tests.-.graphql.txt)

```
[INFO]   #  failure         detail                                                                            
[INFO]                                                                                                        
[INFO]  1.  AssertionError  Bundle uploaded sucessfully                                                       
[INFO]                      expected 'RECEIVED_BUNDLE' to deeply equal 'SUCCESS'                              
[INFO]                      at assertion:0 in test-script                                                     
[INFO]                      inside "Related content with condition / query / pre_ImportBundleWithContext"     
[INFO]                                                                                                        
[INFO]  2.  AssertionError  Bundle uploaded sucessfully                                                       
[INFO]                      expected 'RECEIVED_BUNDLE' to deeply equal 'SUCCESS'                              
[INFO]                      at assertion:0 in test-script                                                     
[INFO]                      inside "Test Render Content Fields / pre_ImportBundleWithContentWithBadVelocity"  
[INFO]                                                                                                        
[INFO]  3.  TypeError       When error rendering field value then original field value is returned            
[INFO]                      Cannot read properties of undefined (reading 'body')                              
[INFO]                      at assertion:0 in test-script                                                     
[INFO]                      inside "Test Render Content Fields / Render content with field with bad velocity" 
[INFO] Collection GraphQLTests had failures. Details:
[INFO] Failure #1: SerialisedError {
[INFO]   name: 'AssertionError',
[INFO]   index: 0,
[INFO]   test: 'Bundle uploaded sucessfully',
[INFO]   message: "expected 'RECEIVED_BUNDLE' to deeply equal 'SUCCESS'",
[INFO]   stack: "AssertionError: expected 'RECEIVED_BUNDLE' to deeply equal 'SUCCESS'\n" +
[INFO]     '   at Object.eval sandbox-script.js:1:70)',
[INFO]   checksum: '7013ea112afd2ef7e72c994693327505',
[INFO]   id: '48b5df53-49b3-4e4a-9da1-784d0e9f4255',
[INFO]   timestamp: 1714190358682,
[INFO]   stacktrace: [
[INFO]     {
[INFO]       fileName: 'sandbox-script.js',
[INFO]       lineNumber: 1,
[INFO]       functionName: 'Object.eval',
[INFO]       typeName: 'Object',
[INFO]       methodName: 'eval',
[INFO]       columnNumber: 70,
[INFO]       native: false
[INFO]     }
[INFO]   ]
[INFO] }
[INFO] Failure #2: SerialisedError {
[INFO]   name: 'AssertionError',
[INFO]   index: 0,
[INFO]   test: 'Bundle uploaded sucessfully',
[INFO]   message: "expected 'RECEIVED_BUNDLE' to deeply equal 'SUCCESS'",
[INFO]   stack: "AssertionError: expected 'RECEIVED_BUNDLE' to deeply equal 'SUCCESS'\n" +
[INFO]     '   at Object.eval sandbox-script.js:1:76)',
[INFO]   checksum: '25ea609b9471e0e659a7244e62f07451',
[INFO]   id: 'bc8a88fd-c464-4fc4-9727-5206b4ce2d7e',
[INFO]   timestamp: 1714190398296,
[INFO]   stacktrace: [
[INFO]     {
[INFO]       fileName: 'sandbox-script.js',
[INFO]       lineNumber: 1,
[INFO]       functionName: 'Object.eval',
[INFO]       typeName: 'Object',
[INFO]       methodName: 'eval',
[INFO]       columnNumber: 76,
[INFO]       native: false
[INFO]     }
[INFO]   ]
[INFO] }
```

Looks like the underlying cause may be linked to an issue scheduling a job when finalizing the commit listeners in the bundle/sync this causes the transaction to fail.

```
DOTCMS:22:03:04.993  INFO  quartz.DotStatefulJob - New Task for job `CleanUpFieldReferencesJob` scheduled for execution `04 03 22 2 5 ? 2024`.
DOTCMS:22:03:05.016  INFO  job.CleanUpFieldReferencesJob - CleanUpFieldReferencesJob ::: started for field `fields0`.
DOTCMS:22:03:05.021  ERROR db.HibernateUtil - Based on configured schedule, the given trigger will never fire.
DOTCMS:com.dotmarketing.business.DotStateException: Based on configured schedule, the given trigger will never fire.
DOTCMS:	at com.dotmarketing.db.DotRunnableThread.run(DotRunnableThread.java:56) ~[classes/:?]
DOTCMS:	at com.dotmarketing.db.HibernateUtil.finalizeCommitListeners(HibernateUtil.java:1165) ~[classes/:?]
DOTCMS:	at com.dotmarketing.db.HibernateUtil.commitTransaction(HibernateUtil.java:1257) ~[classes/:?]
DOTCMS:	at com.dotcms.publisher.receiver.BundlePublisher.process(BundlePublisher.java:234) ~[classes/:?]
DOTCMS:	at com.dotcms.rest.PushPublisherJob.processBundle(PushPublisherJob.java:40) ~[classes/:?]
DOTCMS:	at com.dotcms.rest.BundleResource.uploadBundleSync(BundleResource.java:892) ~[classes/:?]
DOTCMS:	at jdk.internal.reflect.GeneratedMethodAccessor2016.invoke(Unknown Source) ~[?:?]
DOTCMS:	at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:?]
DOTCMS:	at java.lang.reflect.Method.invoke(Method.java:566) ~[?:?]
DOTCMS:	at org.glassfish.jersey.server.model.internal.ResourceMethodInvocationHandlerFactory$1.invoke(ResourceMethodInvocationHandlerFactory.java:81) ~[jersey-server-2.22.1.jar:?]
DOTCMS:	at org.glassfish.jersey.server.model.internal.AbstractJavaResourceMethodDispatcher$1.run(AbstractJavaResourceMethodDispatcher.java:144) ~[jersey-server-2.22.1.jar:?]
DOTCMS:	at org.glassfish.jersey.server.model.internal.AbstractJavaResourceMethodDispatcher.invoke(AbstractJavaResourceMethodDispatcher.java:161) ~[jersey-server-2.22.1.jar:?]
DOTCMS:	at org.glassfish.jersey.server.model.internal.JavaResourceMethodDispatcherProvider$ResponseOutInvoker.doDispatch(JavaResourceMethodDispatcherProvider.java:160) ~[jersey-server-2.22.1.jar:?]
DOTCMS:	at org.glassfish.jersey.server.model.internal.AbstractJavaResourceMethodDispatcher.dispatch(AbstractJavaResourceMethodDispatcher.java:99) ~[jersey-server-2.22.1.jar:?]
DOTCMS:	at org.glassfish.jersey.server.model.ResourceMethodInvoker.invoke(ResourceMethodInvoker.java:389) ~[jersey-server-2.22.1.jar:?]
DOTCMS:	at org.glassfish.jersey.server.model.ResourceMethodInvoker.apply(ResourceMethodInvoker.java:347) ~[jersey-server-2.22.1.jar:?]
DOTCMS:	at org.glassfish.jersey.server.model.ResourceMethodInvoker.apply(ResourceMethodInvoker.java:102) ~[jersey-server-2.22.1.jar:?]
DOTCMS:	at org.glassfish.jersey.server.ServerRuntime$2.run(ServerRuntime.java:326) ~[jersey-server-2.22.1.jar:?]
DOTCMS:	at org.glassfish.jersey.internal.Errors$1.call(Errors.java:271) ~[jersey-common-2.22.1.jar:?]
DOTCMS:	at org.glassfish.jersey.internal.Errors$1.call(Errors.java:267) ~[jersey-common-2.22.1.jar:?]
DOTCMS:	at org.glassfish.jersey.internal.Errors.process(Errors.java:315) ~[jersey-common-2.22.1.jar:?]
DOTCMS:	at org.glassfish.jersey.internal.Errors.process(Errors.java:297) ~[jersey-common-2.22.1.jar:?]
DOTCMS:	at org.glassfish.jersey.internal.Errors.process(Errors.java:267) ~[jersey-common-2.22.1.jar:?]
DOTCMS:	at org.glassfish.jersey.process.internal.RequestScope.runInScope(RequestScope.java:317) ~[jersey-common-2.22.1.jar:?]
DOTCMS:	at org.glassfish.jersey.server.ServerRuntime.process(ServerRuntime.java:305) ~[jersey-server-2.22.1.jar:?]
DOTCMS:	at com.dotmarketing.db.HibernateUtil.finalizeCommitListeners(HibernateUtil.java:1165) ~[classes/:?]
DOTCMS:	at com.dotmarketing.db.HibernateUtil.commitTransaction(HibernateUtil.java:1257) ~[classes/:?]
DOTCMS:	... 102 more
DOTCMS:Caused by: org.quartz.SchedulerException: Based on configured schedule, the given trigger will never fire.
DOTCMS:	at org.quartz.core.QuartzScheduler.rescheduleJob(QuartzScheduler.java:1011) ~[dot.quartz-all-1.8.6_2.jar:?]
DOTCMS:	at org.quartz.impl.StdScheduler.rescheduleJob(StdScheduler.java:302) ~[dot.quartz-all-1.8.6_2.jar:?]
DOTCMS:	at com.dotmarketing.quartz.QuartzUtils.scheduleTask(QuartzUtils.java:494) ~[classes/:?]
DOTCMS:	at com.dotmarketing.quartz.DotStatefulJob.enqueueTrigger(DotStatefulJob.java:171) ~[classes/:?]
DOTCMS:	at com.dotmarketing.quartz.job.CleanUpFieldReferencesJob.lambda$triggerCleanUpJob$0(CleanUpFieldReferencesJob.java:118) ~[classes/:?]
DOTCMS:	at com.rainerhahnekamp.sneakythrow.Sneaky.lambda$sneaked$7(Sneaky.java:207) ~[sneakythrow-1.1.0.jar:?]
DOTCMS:	at com.dotmarketing.db.DotRunnableThread.internalRunner(DotRunnableThread.java:99) ~[classes/:?]
DOTCMS:	at com.dotmarketing.db.LocalTransaction.wrap(LocalTransaction.java:165) ~[classes/:?]
DOTCMS:	at com.dotmarketing.db.DotRunnableThread.run(DotRunnableThread.java:53) ~[classes/:?]
DOTCMS:	at com.dotmarketing.db.HibernateUtil.finalizeCommitListeners(HibernateUtil.java:1165) ~[classes/:?]
DOTCMS:	at com.dotmarketing.db.HibernateUtil.commitTransaction(HibernateUtil.java:1257) ~[classes/:?]
DOTCMS:	... 102 more
[INFO]   POST http://localhost:8080/api/bundle/sync [200 OK, 1.09kB, 5.1s]
DOTCMS:22:03:05.073  ERROR Failed to publish because an error occurred:  - Error publishing bundle with ID 'pageDisplayingVisitor.tar.gz': com.dotmarketing.exception.DotHibernateException: Unable to close Hibernate Session
DOTCMS:22:03:05.073  INFO  util.PushPublishLogger - class com.dotcms.rest.PushPublisherJob : [BundleID: pageDisplayingVisitor.tar.gz] Finished bundle publish process
DOTCMS:22:03:05.073  INFO  rest.PushPublisherJob - Finished bundle publish process
```

--- a/dotCMS/src/main/java/com/dotmarketing/quartz/DotStatefulJob.java
+++ b/dotCMS/src/main/java/com/dotmarketing/quartz/DotStatefulJob.java
@@ -157,19 +157,16 @@ public abstract class DotStatefulJob extends DotJob implements StatefulJob {
             triggersData.put(nextTriggerName, nextExecutionData);
             jobProperties.put(TRIGGER_JOB_DETAIL, triggersData);
 
-            final Calendar calendar = Calendar.getInstance();
-            final String cronString = new SimpleDateFormat("ss mm H d M ? yyyy")
-                    .format(calendar.getTime());
-            final ScheduledTask task = new CronScheduledTask(jobName,
+            final ScheduledTask task = new SimpleScheduledTask(jobName,
                     groupName, description,
-                    jobClass.getCanonicalName(), false,
+                    jobClass.getCanonicalName(),false,
                     nextTriggerName, triggerGroup, new Date(), null,
-                    SimpleTrigger.MISFIRE_INSTRUCTION_FIRE_NOW, 10, jobProperties,
-                    cronString);
+                    SimpleTrigger.MISFIRE_INSTRUCTION_FIRE_NOW, 10, false, jobProperties,0,0);
+
             task.setDurability(true); //must be durable to preserve the detail across triggers.
 
             QuartzUtils.scheduleTask(task);
-            Logger.info(DotStatefulJob.class, String.format("New Task for job `%s` scheduled for execution `%s`.", jobName, cronString));
+            Logger.info(DotStatefulJob.class, String.format("New Task for job `%s` scheduled`.", jobName));
 
         }
     }

--- a/dotCMS/src/main/java/com/dotmarketing/quartz/job/CleanUpFieldReferencesJob.java
+++ b/dotCMS/src/main/java/com/dotmarketing/quartz/job/CleanUpFieldReferencesJob.java
@@ -109,14 +109,23 @@ public class CleanUpFieldReferencesJob extends DotStatefulJob {
 
     public static void triggerCleanUpJob(final Field field, final User user) {
 
-        final Map<String, Serializable> nextExecutionData = ImmutableMap
+        final Map<String, Serializable> nextExecutionData = Map
                 .of("field", field,
                         "deletionDate", Calendar.getInstance().getTime(),
                         "user", user);
 
-        HibernateUtil.addCommitListenerNoThrow(Sneaky.sneaked(() -> {
-                    enqueueTrigger(nextExecutionData, CleanUpFieldReferencesJob.class);
-                }
-        ));
+        HibernateUtil.addCommitListenerNoThrow(
+                () -> {
+                    try {
+                        enqueueTrigger(nextExecutionData, CleanUpFieldReferencesJob.class);
+                    } catch (final Exception e) {
+                        // Do not fail the transaction if the job can't be enqueued
+                        Logger.error(
+                                CleanUpFieldReferencesJob.class,
+                                "Error enqueuing CleanUpFieldReferencesJob for field "
+                                        + field.variable(),
+                                e);
+                    }
+                });
     }
 }

